projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -6245,7 +6235,12 @@ export abstract class IgxGridBaseDirective implements GridType,
     // TODO: do not remove this, as it is used in rowEditTemplate, but mark is as internal and hidden
     /* blazorCSSuppress */
     public endEdit(commit = true, event?: Event): boolean {
-        return this.crudService.endEdit(commit, event);
+        const success = this.crudService.endEdit(commit, event);
+        // simulate the process of focusing out of the cell so that the active element is cleared 
+        // while this should naturally happen when clicking a button (as a most likely endEdit called)
+        // the intentional focusout ignore during edit mode prevents the expected behavior
+        this.onFocusOut({ target: this.tbody.nativeElement } as unknown as FocusEvent);
```

Comment:
Eh, I think it might be a bit too risky to assume all calls to this API are coming when the Grid is focused out and clear selection.
Actually, there's a case for the opposite that currently has an issue. See Row Edit sample in dev demos, check Custom Template (it should have been using `endEdit()` as exposed by the template context, but I digress). Point being the buttons on the row edit overlay are intended to call into `endEdit` but loose focus breaking KB nav. Compare w/ what the default template use - the internal `endRowEditTabStop` that checks for active node and returns focus to the grid tbody. Note that check is also silly and fails if the last active node is header, etc.

My thinking is for this method to handle both cases correctly, it should probably check the focused state of the grid before restoring focus or clearing active cell. Something in the lines of:
```ts
        const activeCell = this.navigation.activeNode;
        const document = this.nativeElement?.getRootNode() as Document | ShadowRoot;

        if (!activeCell) {
            return success;
        }

        if (this.nativeElement?.contains(document.activeElement) && activeCell.row !== -1) {
            // restore focus for navigation, TODO: navigation method to handle all cases...
            this.tbody.nativeElement.focus();
        } else {
            // grid already lost focus, clear active
            this.navigation.lastActiveNode = this.navigation.activeNode;
            this.navigation.activeNode = {} as IActiveNode;
            this.notifyChanges();
        }
```
This else can be refactored to be shared with the onFocusOut handler, but I see no reason for the additional checks - if the grid lost focus prior to end edit, might as well cleanup the active node that editing was blocking.

Reply:
Oh, okay, so the logic to restore the correct focus is actually baked into the focus out checks :)
```ts
if (activeNode.row >= 0 && activeNode.row < this.dataView.length) this.tbody.nativeElement.focus()
if (activeNode.row === -1) this.theadRow.nativeElement.focus();
if (activeNode.row === this.dataView.length) this.tfoot.nativeElement.focus();
```
eh 😶